jax/_src/mesh.py

```diff
@@ -304,7 +304,11 @@ def empty(self):
 
   @functools.cached_property
   def is_multi_process(self):
-    return self.devices.size != len(self.local_devices)
+    return len(self.process_ids) > 1
+
+  @functools.cached_property
+  def process_ids(self):
```

Comment:
maybe make this return a tuple?

(as discussed offline, it would be good to see the big CL to get the full picture. Then it would easy to approve these small PRs)